File: fluortenten.py
n = int(input())

l = [x for x in map(int, input().split(' ')) if x != 0]
        
sum = 0
acc = 0
best = 0
        
for i, v in enumerate(reversed(l)):
    acc += v
    sum += (len(l)-i)*v
    
    if acc > best:
        best = acc

print(best+sum)
    

File: fluortenten.cpp
#include<iostream>
#include<vector>
#include<algorithm>

using namespace std;

int main() {
    
    int n, i, tmp;

    cin >> n;

    vector<int> vec;

    while (cin >> tmp) {
        if (tmp != 0) {
            vec.push_back(tmp);
        }
    }

    int best = 0;
    int acc = 0;
    int sum = 0;

    reverse(vec.begin(), vec.end());

    for (i=0; i<vec.size(); i++) {
        acc += vec[i];
        sum += (vec.size()-i)*vec[i];

        if (acc > best) {
            best = acc;
        }
    }

    cout << (sum+best) << endl;

    return 0;
}